api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java

```diff
@@ -181,9 +188,26 @@ private Claim buildClaim(ClaimStatus claimStatus, ClaimRequest claimRequest, Eli
                 .initialIdentityAndEligibilityResponse(decision.getIdentityAndEligibilityResponse())
                 .currentIdentityAndEligibilityResponse(decision.getIdentityAndEligibilityResponse())
                 .eligibilityOverride(claimRequest.getEligibilityOverride())
+                .reference(reference)
                 .build();
     }
 
+    private String checkAndReturnUniqueReferenceForClaim() {
+        String reference;
+        boolean isReferenceDuplicate;
+        do {
+            reference = ReferenceGenerator.generateReference(claimReferenceSize);
+            isReferenceDuplicate = false;
```

Comment:
This should be false by default so no need to assign

Reply:
yeah i did think about that but if we find duplicate reference then it will never reset to false and might end up in infinite while loop.